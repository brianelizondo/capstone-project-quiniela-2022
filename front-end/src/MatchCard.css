.MatchCard-card{
    filter: drop-shadow(5px 5px 5px #000);
    margin-bottom: 50px;
}
.MatchCard-match{
    font-size: 15px;
    font-weight: 700;
    text-align: center;
    text-transform: uppercase;
}
.MatchCard-group{
    color: gray;
    font-size: 13px;
    text-align: center;
    margin-top: -5px;
}
.MatchCard-teamlogo{
    text-align: center !important;
}
.MatchCard-teamlogo img{
    aspect-ratio: auto;
    max-height: 100px;
    max-width: 90px;
}
.MatchCard-teamnameA, .MatchCard-teamnameB{
    font-weight: 700;
    font-size: 18px;
    text-transform: uppercase;
    color: black;
    padding-top: 25px;
}
.MatchCard-teamnameA{
    text-align: left;
}
.MatchCard-teamnameB{
    text-align: right;
}
.MatchCard-teamgoalsA, .MatchCard-teamgoalsB{
    color: gray;
    font-size: 11px;
    white-space: pre;
}
.MatchCard-teamgoalsA{
    text-align: left;
}
.MatchCard-teamgoalsB{
    text-align: right;
}
.MatchCard-result{
    font-family: 'Rubik', sans-serif;
    font-size: 50px;
    font-weight: 700;
    text-align: center;
    text-transform: uppercase;
}


.MatchCard-date, .MatchCard-time, .MatchCard-location{
    font-size: 11px;
    text-align: center;
}