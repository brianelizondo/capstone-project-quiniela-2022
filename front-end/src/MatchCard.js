import React from "react";
import { Link } from "react-router-dom";
import { Card, Image, Row, Col } from 'react-bootstrap';

function MatchCard({ match }){
    let phase;
    let teamAName;
    let teamALogo = "";
    let teamBName;
    let teamBLogo = "";
    if(match.id <= 48){
        phase = 1;
        teamAName = match.teamA.name;
        teamALogo = <Image src={ `/images/team_logo/${match.teamA.shortName}.png` } alt={ match.teamA.name } fluid={true} />;
        teamBName = match.teamB.name;
        teamBLogo = <Image src={ `/images/team_logo/${match.teamB.shortName}.png` } alt={ match.teamB.name } fluid={true} />;
    }else{
        phase = 2;
        if(match.teamA.id > 0){
            teamAName = match.teamA.name;
            teamALogo = <Image src={ `/images/team_logo/${match.teamA.shortName}.png` } alt={ match.teamA.name } fluid={true} />;
        }else{
            teamAName = match.teamA_classified;
        }
        if(match.teamB.id > 0){
            teamBName = match.teamB.name;
            teamBLogo = <Image src={ `/images/team_logo/${match.teamB.shortName}.png` } alt={ match.teamA.name } fluid={true} />;
        }else{
            teamBName = match.teamB_classified;
        }
    }

    const matchResult = match.teamA_result !== null && match.teamB_result !== null ? `${match.teamA_result} - ${match.teamB_result}` : "vs";
        
    return (
        <Link key={match.id} to={`/matches/phase/${ phase }/match/${match.id}`}>
            <Card data-testid={`match-${match.id}`}>
                <Card.Body>
                    <Row>
                        <Col>Match { match.id }</Col>
                    </Row>
                    <Row>
                        <Col xs lg="2">{ teamALogo }</Col>
                        <Col>{ teamAName }</Col>
                        <Col>{ matchResult }</Col>
                        <Col>{ teamBName }</Col>
                        <Col xs lg="2">{ teamBLogo }</Col>
                    </Row>
                    <Row>
                        <Col>{ match.date }</Col>
                    </Row>
                    <Row>
                        <Col>{ match.time }</Col>
                    </Row>
                    <Row>
                        <Col>{ match.city } - { match.stadium }</Col>
                    </Row>
                </Card.Body>
            </Card>
        </Link>
    );
}

export default MatchCard;