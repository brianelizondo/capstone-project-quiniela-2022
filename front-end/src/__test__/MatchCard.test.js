import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import MatchCard from '../MatchCard';

const match = {
    id: 1,
    date: "11/20/2022",
    time: "11:00",
    stadium: "stadium",
    city: "city",
    group: "A",
    teamA: {
        id: 1,
        name: "Qatar",
        shortName: "QAT",
        apiID: 1569
    },
    teamA_result: 0,
    teamB: {
        id: 2,
        name: "Ecuador",
        shortName: "ECU",
        apiID: 2382
    },
    teamB_result: 0,
    status: 1,
    apiID: 855736
}

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'MatchCard' component without crashing", () => {
        render(
            <MemoryRouter>
                <MatchCard match={match} />
            </MemoryRouter>
        );
        const textFound = screen.getByText("Match 1");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'MatchCard' matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter>
                <MatchCard match={match} />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});