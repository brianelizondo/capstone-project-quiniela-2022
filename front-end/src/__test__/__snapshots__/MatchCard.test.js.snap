// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'MatchCard' matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/matches/phase/1/match/1"
  >
    <div
      class="card"
      data-testid="match-1"
    >
      <div
        class="card-body"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            Match 1
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col-lg-2 col"
          >
            <img
              alt="Qatar"
              class="img-fluid"
              src="/images/team_logo/QAT.png"
            />
          </div>
          <div
            class="col"
          >
            Qatar
          </div>
          <div
            class="col"
          >
            0 - 0
          </div>
          <div
            class="col"
          >
            Ecuador
          </div>
          <div
            class="col-lg-2 col"
          >
            <img
              alt="Ecuador"
              class="img-fluid"
              src="/images/team_logo/ECU.png"
            />
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            11/20/2022
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            11:00
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            city - stadium
          </div>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
